### ISSUE TITLE
Add value_and_grad option to all gradient-based solvers

### ISSUE THREAD

**Issue by mblondel** [google/jaxopt#115]
This will allow the user to override the default behavior (jax.value_and_grad).
